src/mapgen/mapgen.cpp

```diff
@@ -870,29 +870,31 @@ void MapgenBasic::generateDungeons(s16 max_stone_y)
 	if (num_dungeons == 0)
 		return;
 
-	// Get biome at mapchunk midpoint
-	v3s16 chunk_mid = node_min + (node_max - node_min) / v3s16(2, 2, 2);
-	Biome *biome = (Biome *)biomegen->getBiomeAtPoint(chunk_mid);
-
+	PseudoRandom ps(blockseed + 70033);
+	
 	DungeonParams dp;
 
+	dp.np_alt_wall =
+		NoiseParams(-0.4, 1.0, v3f(40.0, 40.0, 40.0), 32474, 6, 1.1, 2.0);
+
 	dp.seed             = seed;
+	dp.num_dungeons     = num_dungeons;
 	dp.only_in_ground   = true;
+	dp.num_rooms        = ps.range(2, 16);
+	dp.room_size_min    = v3s16(6, 5, 6);
+	dp.room_size_max    = v3s16(10, 6, 10);
+	dp.room_size_large  = v3s16(
+		ps.range(10, 18), ps.range(8, 16), ps.range(10, 18));
+	dp.first_room_large = ps.range(1, 4) == 1;
+	dp.holesize         = v3s16(2, 3, 2);
 	dp.corridor_len_min = 1;
 	dp.corridor_len_max = 13;
-	dp.rooms_min        = 2;
-	dp.rooms_max        = 16;
+	dp.diagonal_dirs    = ps.range(1, 12) == 1;
```

Comment:
We can no longer have diagonal corriodrs in desert dungeons (because that would be hardcoded), so now the chance of diagonal corridors is made 1 in 12 for all biomes. The chance is chosen because desert is 1 out of the 12 MTG biomes.